extensions/ql-vscode/src/remote-queries/gh-actions-api-client.ts

```diff
@@ -142,19 +207,22 @@ async function downloadArtifact(
     archive_format: 'zip',
   });
   const artifactPath = path.join(tmpDir.name, `${artifactId}`);
-  void logger.log(`Downloading artifact to ${artifactPath}.zip`);
-  await fs.writeFile(
-    `${artifactPath}.zip`,
-    Buffer.from(response.data as ArrayBuffer)
-  );
-
-  void logger.log(`Extracting artifact to ${artifactPath}`);
-  await (
-    await unzipper.Open.file(`${artifactPath}.zip`)
-  ).extract({ path: artifactPath });
+  await saveFile(`${artifactPath}.zip`, response.data as ArrayBuffer);
+  await unzipFile(`${artifactPath}.zip`, artifactPath);
   return artifactPath;
 }
 
+async function saveFile(filePath: string, data: ArrayBuffer): Promise<void> {
+  void logger.log(`Saving file to ${filePath}`);
+  await fs.writeFile(filePath, Buffer.from(data));
+}
+
+async function unzipFile(sourcePath: string, destinationPath: string) {
+  void logger.log(`Unzipping file to ${destinationPath}`);
+  const file = await unzipper.Open.file(sourcePath);
+  await file.extract({ path: destinationPath });
```

Comment:
Careful here...unzipping can be tricky sometimes. There are certain zip files that cannot be extracted this way. See the workaround here: https://github.com/github/vscode-codeql/blob/1014c4bdda3424d3dd8b586e44391b742c1ceaa4/extensions/ql-vscode/src/databaseFetcher.ts#L285-L289

What you have _might_ work, but then again, it may fail unexpectedly on some larger files.

Reply:
Arrgh...sorry. I think what you have is fine. I just wasn't reading it carefully. I'm not particularly happy with this library as we've had so many problems with it in the past. Particularly, large zips (> 2GB) can't be handled by the library. But, that isn't a problem here.

Reply:
That is good to know, thank you!